diff --git a/arch/arm/dts/rk3528-u-boot.dtsi b/arch/arm/dts/rk3528-u-boot.dtsi
index eb6a55cd5c9..bb681fbcfa4 100644
--- a/arch/arm/dts/rk3528-u-boot.dtsi
+++ b/arch/arm/dts/rk3528-u-boot.dtsi
@@ -46,6 +46,22 @@
 			status = "disabled";
 		};
 	};
+
+	i2c1: i2c@ffa58000 {
+        compatible = "rockchip,rk3528-i2c", "rockchip,rk3399-i2c";
+        reg = <0x0 0xffa58000 0x0 0x1000>;
+        interrupts = <GIC_SPI 62 IRQ_TYPE_LEVEL_HIGH>;
+        clocks = <&cru CLK_I2C1>, <&cru PCLK_I2C1>;
+        clock-names = "i2c", "pclk";
+        //resets = <&cru SRST_I2C1>;
+        probed = <1>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&i2c1m0_uboot_xfer>;
+        clock-frequency = <100000>;
+        status = "okay";
+        u-boot,dm-pre-reloc;
+        bootph-all;
+    };
 };
 
 &cru {
@@ -99,6 +115,12 @@
 
 &pinctrl {
 	bootph-all;
+	i2c1m0_uboot_xfer: i2c1m0-uboot-xfer {
+		u-boot,dm-pre-reloc;
+		rockchip,pins =
+			<4 RK_PA3 2 &pcfg_pull_none_smt>,  /* I2C1_SCL_M0 */
+			<4 RK_PA2 2 &pcfg_pull_none_smt>;  /* I2C1_SDA_M0 */
+	};
 };
 
 &sdhci {
